lib/hls/hls_parser.js

```diff
@@ -1893,9 +1893,20 @@ shaka.hls.HlsParser = class {
         const pPreviousReference = i == 0 ?
           previousReference : partialSegmentRefs[partialSegmentRefs.length - 1];
         const pStartTime = (i == 0) ? startTime : pPreviousReference.endTime;
-        const pDuration = Number(item.getAttributeValue('DURATION'));
-        // A preload hinted partial segment doesn't have duration information,
-        // so its startTime and endTime are the same.
+
+        // If DURATION is missing from this partial segment, use the target
+        // partial duration from the top of the playlist, which is a required
+        // attribute for content with partial segments.
+        const pDuration = Number(item.getAttributeValue('DURATION')) ||
```

Comment:
Technically this approach means that if a manifest has a tag with `DURATION=0`, we will ignore that. But I don't know why someone would want to do that, so it's probably fine to not account for it.